shell/components/ButtonDropdown.vue

```diff
@@ -211,14 +211,15 @@ export default {
     <!-- Pass down templates provided by the caller -->
     <template
       v-for="(_, slot) of $slots"
+      #[slot]="scope"
       :key="slot"
-      v-slot:[slot]="scope"
     >
-      <slot
-        v-if="slot !== 'selected-option'"
-        :name="slot"
-        v-bind="scope"
-      />
+      <template v-if="slot !== 'selected-option' && typeof $slots[slot] === 'function'">
```

Comment:
Just for my own info, what other values would `typeof $slots[slot]` produce other than a function. Does it just differentiate between plain elements and components?

Reply:
When working on a demonstration, I couldn't get this to evaluate to anything other than `function` and the migration guide docs support this[^1]. 

When I was working on aedd3348035cebaeeea57832e5947f780511fdcd, I recall encountering instances of `undefined`/`string`. This could have been added in error while testing - I will review these cases and follow up in another PR if this line is not needed.

[^1]: https://v3-migration.vuejs.org/breaking-changes/slots-unification.html